cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java

```diff
@@ -567,6 +535,44 @@ void setReplicaLoad(TopicPartition tp, AggregatedMetricValues aggregatedMetricVa
     _load.addMetricValues(aggregatedMetricValues, windows);
   }
 
+  /**
+   * Populate the information of replica placement over disk for the broker.
+   *
+   * @param logdir The logdir of the disk.
+   * @param replicaInfos the replicas hosted by this disk.
+   */
+  public void populateDiskInfo(String logdir, Map<TopicPartition, DescribeLogDirsResponse.ReplicaInfo> replicaInfos) {
+    Disk disk = disk(logdir);
+    replicaInfos.forEach((key, value) -> {
+      try {
+        if (!value.isFuture) {
+          disk.addReplica(replica(key), true);
+        }
+      } catch (NullPointerException e) {
+        // Let it go.
```

Comment:
This means CapacityResolver does not get the capacity of the disk, in this case we ignore the replica on this disk(since we cannot get disk utilization percentage) and only consider to move replicas between other disks with complete information.